include/networkit/graph/Graph.hpp

```diff
@@ -1363,7 +1363,155 @@ class Graph final {
+                    // undirected
+                    inEdgeWeights = G.inEdgeWeights;
+                    outEdgeWeights = G.outEdgeWeights;
+                } else {
+                    // G has no weights, set defaultEdgeWeight for all edges
+                    if (directed) {
+                        inEdgeWeights.resize(z);
+                        for (node u = 0; u < z; u++) {
+                            inEdgeWeights[u] =
+                                std::vector<edgeweight>(G.inEdges[u].size(), defaultEdgeWeight);
+                        }
+                    }
+
+                    outEdgeWeights.resize(z);
+                    for (node u = 0; u < z; u++) {
+                        outEdgeWeights[u] =
+                            std::vector<edgeweight>(outEdges[u].size(), defaultEdgeWeight);
+                    }
+                }
+            }
+            if (G.hasEdgeIds() && edgesIndexed) {
+                inEdgeIds = G.inEdgeIds;
+                outEdgeIds = G.outEdgeIds;
+            }
+        } else if (G.isDirected()) {
+            // G is directed, but we want an undirected graph
+            // so we need to combine the out and in stuff for every node
+            outEdges.resize(z);
+            if (weighted)
+                outEdgeWeights.resize(z);
+            if (G.hasEdgeIds() && edgesIndexed)
+                outEdgeIds.resize(z);
+            for (node u = 0; u < z; u++) {
+                // copy both out and in edges into our new outEdges
+                outEdges[u].reserve(G.outEdges[u].size() + G.inEdges[u].size());
+                outEdges[u].insert(outEdges[u].end(), G.outEdges[u].begin(), G.outEdges[u].end());
+                if (weighted) {
+                    if (G.isWeighted()) {
+                        // same for weights
+                        outEdgeWeights[u].reserve(G.outEdgeWeights[u].size()
+                                                  + G.inEdgeWeights[u].size());
+                        outEdgeWeights[u].insert(outEdgeWeights[u].end(),
+                                                 G.outEdgeWeights[u].begin(),
+                                                 G.outEdgeWeights[u].end());
+                    } else {
+                        // we are undirected, so no need to write anything into inEdgeWeights
+                        outEdgeWeights[u] =
+                            std::vector<edgeweight>(outEdges[u].size(), defaultEdgeWeight);
+                    }
+                }
+                if (G.hasEdgeIds() && edgesIndexed) {
+                    // copy both out and in edges ids into our new outEdgesIds
+                    outEdgeIds[u].reserve(G.outEdgeIds[u].size() + G.inEdgeIds[u].size());
+                    outEdgeIds[u].insert(outEdgeIds[u].end(), G.outEdgeIds[u].begin(),
+                                         G.outEdgeIds[u].end());
+                }
+            }
+            for (node u = 0; u < z; ++u) {
```

Comment:
Can we use a parallel loop here? This operation seems quite expensive and we only modify the out-edges of `u`.
```suggestion
            G.parallelForNodes([&](node u) {
```